### DISCUSSION TITLE
feedback: cloudflare logs

### DISCUSSION THREAD

**Discussion by danbars** [cloudflare/workers-sdk#7444] [General]
Hi, I have some feedback on Cloudflare Logs, hope it's a good place to publish it - I couldn't find any other reasonable place :)
First think first, it is an awesome addition, really simplifies my code since I no longer have to write custom tail workers to send logs to some third-party service.

2 things about the UI that would really improve the experience, imo:
1. The search box with the other buttons around it should remain fixed at the top, and not scroll with the content. Otherwise you have to scroll all the way up if you want to change the filter
2. When you search or filter, and find the line that you were looking for, there should be a way to locate that line in an unfiltered view. Something like "jump to this line" button. When troubleshooting, looking at one specific line isn't enough. You have to look at the lines that come before or after it. This is super important. 

Thanks!

**Comment by emily-shen** [comment]
cc @Ankcorn

**Comment by Ankcorn** [comment]
Thanks Dan!

> 1. The search box with the other buttons around it should remain fixed at the top, and not scroll with the content. Otherwise you have to scroll all the way up if you want to change the filter

We are working on this. :)

> 2. When you search or filter, and find the line that you were looking for, there should be a way to locate that line in an unfiltered view. Something like "jump to this line" button. When troubleshooting, looking at one specific line isn't enough. You have to look at the lines that come before or after it. This is super important.

We have a feature on our roadmap to find logs that matched a certain filter and then see all the logs for that request grouped together. Would this solve that problem for you?

**Reply by danbars** [reply]
> find logs that matched a certain filter and then see all the logs for that request grouped together. Would this solve that problem for you?

This will solve many cases, I believe, where you wanth to watch things related to the same request. 
However, I think it's still needed to be able to see the unfiltered log, and jump to the specific line.
This is because some things span multiple requests. For example, multiple requests on durable objects, workflows, and even multiple requests that do some crud operations on a db.
